File: app/src/main/java/com/example/bilibili/MainActivity.java
package com.example.bilibili;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;

import com.example.bilibili.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    // Used to load the 'bilibili' library on application startup.
    static {
        System.loadLibrary("bilibili");
    }

    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        // Example of a call to a native method
        TextView tv = binding.sampleText;
        tv.setText(stringFromJNI());
    }

    /**
     * A native method that is implemented by the 'bilibili' native library,
     * which is packaged with this application.
     */
    public native String stringFromJNI();
}

File: app/src/main/java/com/mvvm/news/viewmodel/MainActivity.kt
package com.mvvm.news.viewmodel

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.databinding.ObservableField
import androidx.lifecycle.MutableLiveData
import com.example.bilibili.R
import com.example.bilibili.databinding.MvvmActivityMainBinding

/**
Created by Mr.Chan
Time 2024-06-17
Blog https://www.cnblogs.com/Frank-dev-blog/
 */
class MainActivity : AppCompatActivity() {
    val number = MutableLiveData<Int>().apply { value = 0 }
    private lateinit var binding: MvvmActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.mvvm_activity_main)
        binding.viewModel = this
        binding.lifecycleOwner = this
    }

    fun add() {
        val currentNumber = number.value ?: 0
        number.value = currentNumber + 1
    }
}
